sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2Suite.scala

```diff
@@ -393,6 +393,29 @@ class DataSourceV2Suite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-32708: same columns with different ExprIds should be equal after canonicalization ") {
```

Comment:
If we don't have an end-to-end test, how about a low-level UT? Create two `DataSourceV2ScanExec` instances and check `scan1.sameResult(scan2)`.

Reply:
@cloud-fan I think this test case creates two `DataSourceV2ScanExec` and do the check. It looks ok to me.